File: platform-api/app/models/project_asset.py
"""ProjectAsset model — unstructured documents as first-class project assets."""

from __future__ import annotations

from sqlalchemy import JSON, BigInteger, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin

_JSON = JSONB().with_variant(JSON(), "sqlite")


class ProjectAsset(TimestampMixin, Base):
    __tablename__ = "project_assets"

    id: Mapped[int] = mapped_column(primary_key=True)
    tenant_id: Mapped[int] = mapped_column(
        ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False, index=True,
    )
    project_id: Mapped[int] = mapped_column(
        ForeignKey("projects.id", ondelete="CASCADE"), nullable=False, index=True,
    )
    owner_user_id: Mapped[int | None] = mapped_column(
        ForeignKey("users.id", ondelete="SET NULL"), nullable=True,
    )
    asset_type: Mapped[str] = mapped_column(String(100), nullable=False)
    source_type: Mapped[str] = mapped_column(String(100), nullable=False, default="uploaded_file")
    title: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    filename: Mapped[str] = mapped_column(Text, nullable=False)
    original_filename: Mapped[str | None] = mapped_column(Text, nullable=True)
    content_type: Mapped[str | None] = mapped_column(String(255), nullable=True)
    file_extension: Mapped[str | None] = mapped_column(String(50), nullable=True)
    storage_provider: Mapped[str] = mapped_column(String(50), nullable=False, default="local")
    storage_location: Mapped[str] = mapped_column(Text, nullable=False)
    file_hash: Mapped[str | None] = mapped_column(Text, nullable=True)
    file_size_bytes: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    visibility: Mapped[str] = mapped_column(String(50), nullable=False, default="shared_project")
    access_group_id: Mapped[int | None] = mapped_column(nullable=True)
    status: Mapped[str] = mapped_column(String(50), nullable=False, default="uploaded")
    ai_status: Mapped[str] = mapped_column(String(50), nullable=False, default="pending")
    ai_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    ai_metadata: Mapped[dict] = mapped_column(_JSON, nullable=False, default=dict)
    ai_error_message: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_by: Mapped[int | None] = mapped_column(
        ForeignKey("users.id", ondelete="SET NULL"), nullable=True,
    )

    def __repr__(self) -> str:
        return f"ProjectAsset(id={self.id}, title={self.title!r}, asset_type={self.asset_type!r})"

File: web-ui/lib/ui/use-project-data/project-asset.ts
"use client";


import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
import { useMemo } from "react";
import { apiClient } from "@/lib/api-client";
import { useCurrentUser, useProjectSummaries } from "../use-shell-data";
import type {
  PresentationDescriptor,
  ResponseEnvelope,
} from "@/lib/api/ai-actions";
import type {
  CurrentUser,
  ProjectSummary,
  TenantSummary,
} from "../types";


// ── Documents (project assets) ───────────────────────────────────────

export interface ProjectAsset {
  id: number;
  project_id: number;
  asset_type: string;
  source_type: string;
  title: string;
  description: string | null;
  filename: string;
  original_filename: string | null;
  content_type: string | null;
  file_extension: string | null;
  file_size_bytes: number | null;
  visibility: string;
  status: string;
  ai_status: string;
  ai_summary: string | null;
  ai_metadata: Record<string, unknown>;
  created_by: number | null;
  created_at: string;
  updated_at: string;
}